src/infer_data_manager_shm.h

```diff
@@ -159,6 +162,9 @@ class InferDataManagerShm : public InferDataManagerBase {
   size_t output_shm_size_;
   // Map from shared memory key to its starting address and size
   std::unordered_map<std::string, SharedMemoryData> shared_memory_regions_;
+
+ private:
+  CUDARuntimeLibraryManager cuda_runtime_library_manager_{};
```

Comment:
Actually, I might want to change this to a singleton, because I think this class gets instantiated more than once throughout the process lifetime.

Reply:
Nevermind, this class is only created once per process lifetime.